actors/market/tests/market_actor_test.rs

```diff
@@ -1278,6 +1279,295 @@ fn active_deals_multiple_times_with_different_providers() {
+
+        let f = |rt: &mut MockRuntime, d: &mut DealProposal| {
+            add_participant_funds(rt, client_addr, d.client_balance_requirement() - 1);
+            add_provider_funds(
+                rt,
+                d.provider_collateral.clone(),
+                provider_addr,
+                owner_addr,
+                worker_addr,
+            );
+        };
+        assert_deal_failure(false, f, ExitCode::USR_ILLEGAL_ARGUMENT, Ok(()));
+    }
+
+    #[test]
+    fn provider_does_not_have_enough_balance_for_collateral() {
+        let owner_addr = Address::new_id(OWNER_ID);
+        let provider_addr = Address::new_id(PROVIDER_ID);
+        let worker_addr = Address::new_id(WORKER_ID);
+        let client_addr = Address::new_id(CLIENT_ID);
+
+        let f = |rt: &mut MockRuntime, d: &mut DealProposal| {
+            add_participant_funds(rt, client_addr, d.client_balance_requirement());
+            add_provider_funds(
+                rt,
+                d.provider_collateral.clone() - 1,
+                provider_addr,
+                owner_addr,
+                worker_addr,
+            );
+        };
+        assert_deal_failure(false, f, ExitCode::USR_ILLEGAL_ARGUMENT, Ok(()));
+    }
+
+    #[test]
+    fn unable_to_resolve_client_address() {
+        let f = |_rt: &mut MockRuntime, d: &mut DealProposal| {
+            d.client = Address::new_id(1);
+        };
+        assert_deal_failure(true, f, ExitCode::USR_ILLEGAL_ARGUMENT, Ok(()));
+    }
+
+    #[test]
+    fn signature_is_invalid() {
+        let f = |_rt: &mut MockRuntime, _d: &mut DealProposal| {};
+        assert_deal_failure(true, f, ExitCode::USR_ILLEGAL_ARGUMENT, Err(anyhow!("error")));
+    }
+
+    #[test]
+    fn no_entry_for_client_in_locked_balance_table() {
+        let owner_addr = Address::new_id(OWNER_ID);
+        let provider_addr = Address::new_id(PROVIDER_ID);
+        let worker_addr = Address::new_id(WORKER_ID);
+
+        let f = |rt: &mut MockRuntime, d: &mut DealProposal| {
+            add_provider_funds(
+                rt,
+                d.provider_collateral.clone(),
+                provider_addr,
+                owner_addr,
+                worker_addr,
+            );
+        };
+        assert_deal_failure(false, f, ExitCode::USR_ILLEGAL_ARGUMENT, Ok(()));
+    }
+
+    #[test]
+    fn no_entry_for_provider_in_locked_balance_table() {
+        let client_addr = Address::new_id(CLIENT_ID);
+
+        let f = |rt: &mut MockRuntime, d: &mut DealProposal| {
+            add_participant_funds(rt, client_addr, d.client_balance_requirement());
+        };
+        assert_deal_failure(false, f, ExitCode::USR_ILLEGAL_ARGUMENT, Ok(()));
+    }
+
+    #[test]
+    fn bad_piece_cid() {
+        let f = |_rt: &mut MockRuntime, d: &mut DealProposal| {
+            d.piece_cid = make_piece_cid("random cid".as_bytes());
+        };
+        assert_deal_failure(false, f, ExitCode::USR_ILLEGAL_ARGUMENT, Ok(()));
```

Comment:
first param should be true

Reply:
Once you set this to true this test case will fail because ` make_piece_cid("random cid".as_bytes())` will pass validation.  To get this to fail validation you'll need to create a cid with a different prefix, or pass in an undefined cid.